### ISSUE TITLE
Computer freezes when display resolution & lazer resolution mismatch

### ISSUE THREAD

**Issue by Anthony1x** [ppy/osu#23419]
### Type

Other

### Context
I have a very wide monitor (5120x1440). Usually I use the built-in screen scaling feature because of this, however for the purposes of streaming osu! in a discord server, I have set my displays and lazers resolution down to a 16:9 resolution (2560x1440) for a more pleasant viewing experience. I use OpenTabletDriver externally and had to adjust my area outside of lazer, which is to explain why I had changed both my monitor as well as lazer's instead of just lazer.

### The Bug
This was all fine until when I tried to start lazer today and was met with a hard freeze. Not just lazer not starting or similar - my PC was quite dead. Mouse was not moving. All my keyboard shortcut attempts (Alt+F4, Ctrl+Shift+Del, Ctrl+Shift+Esc) fruitless (curiously though sound sources continued to play as usual if present). I had set my monitor back to its native resolution, though lazer was still set to a custom one. 

After a brief amount of trial and error, I have discovered that this crash was a result of the mismatching amount of pixels. After setting my display back to 2560x1440 I was able to start lazer normally again and resolve the issue from there.

### Steps to reproduce
**Make sure you have no unsaved work before you try this!**

1. Set display on which lazer runs to a custom resolution
2. Start lazer
3. Set lazer to the same resolution (Make sure not to use "Default resolution", this will not crash the computer)
4. Close lazer
5. Set monitor back to native resolution
6. Start lazer again
7. Freeze occurs

**NOTE:** Weirdly enough, this does **not** happen when only lazer is set to a custom resolution. It has to be the monitor **as well as** lazer.

### Additional information
- No crash logs are being generated when this issue occurs. I guess my PC crashes before lazer has any time to write logs for that to happen. I've still included all logs that were present though, just in case.
- This does **not** happen if lazers resolution is set to "Default". It has to have been explicitly set to a custom resolution for the freeze to occur.
- This is my first bug report ever. Please tell me if I have missed vital information.

EDIT: I asked a friend to try and reproduce this issue and he was not able to. Very strange, it is consisent for me. Definitely do ask me about specifics. I will append new findings as necessary.

### Version

2023.419.0, .NET 6.0.15

### Logs

[runtime.log](https://github.com/ppy/osu/files/11412774/runtime.log)
[performance.log](https://github.com/ppy/osu/files/11412798/performance.log)
[updater.log](https://github.com/ppy/osu/files/11412799/updater.log)
[database.log](https://github.com/ppy/osu/files/11412800/database.log)
[input.log](https://github.com/ppy/osu/files/11412801/input.log)
[network.log](https://github.com/ppy/osu/files/11412802/network.log)

**Comment by peppy** [comment]
This sounds like a driver level issue. osu! causing your whole PC to crash is nigh impossible.

To confirm it's a driver issue, could you check whether you can still hear game sounds working when it stops responding? If you hit space bar for instance, do year hear the associated sound effects?

**Comment by Anthony1x** [comment]
> could you check whether you can still hear game sounds working when it stops responding?

Computer freezes before osu! is able to make any sound. By the time of the freeze osu! is still on its first frames on a black screen.
However, you are right, it was a driver issue. I performed a clean installation of my nvidia drivers and that actually fixed it.
I've tested this exact thing with some other games but it never occurred, so I reckoned it has to have something to do with lazer. I was wrong.

Thanks for pointing this out, cheers!
